gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -115,6 +159,20 @@ private GoogleHadoopFSInputStream(
         new GhfsStreamStats(storageStatistics, GhfsStatistic.STREAM_READ_SEEK_OPERATIONS, gcsPath);
 
     this.traceFactory = ghfs.getTraceFactory();
+    this.vectoredIO = new VectoredIOImpl(ghfs.getGcsFs(), gcsPath, fileInfo, vectoredReadOptions);
```

Comment:
We can't have VectoredIOImpl as part of FileSystem as we need `fileInfo` and technically it will only be available when `open` is requested by caller.

Reply:
I think this is not the right thing to do. As discussed, why not pass the fileInfo in the readVectored?